proposals/credential-management.md

```diff
@@ -0,0 +1,128 @@
+# browser.credentials
+
+WebExtension proposal to allow interfacing with the [Credential Management API](https://w3c.github.io/webappsec-credential-management)
+
+## Motivation
+
+Password manager extensions currently rely on a mixture of heuristics and API overriding to detect and store credentials.
+To fill credentials they also rely on DOM APIs to detect relevant input fields to display UI (or autofill).
+
+Browsers have APIs for interaction with various credential types, most notably PublicKey and Password (Chromium only). However, these aren't exposed to browser extensions.
+
+## Proposal
+
+We propose a new browser.credentials API that would allow extensions to integrate with these existing web APIs.
+
+A mock for this proposal is available [here](credential-management-mock.js) and the corresponding web_accessible_resource that's required is [here](credential-management-mock-pagescript.js).
+
+## Types
+
+```ts
+interface ExtensionCredentialsContainer {
+    create(options?: CredentialCreationOptions): Promise<Credential | null>;
+    get(options?: CredentialRequestOptions): Promise<Credential | null>;
+    store(credential: Credential): Promise<void>;
+}
+
+// Type of browser.credentials
+interface ExtensionCredentialManagement {
+	registerContainer(credentialsContainer: ExtensionCredentialsContainer): Promise<void>;
+	isActiveContainer(): boolean;
+}
+```
+
+### API
+
+**browser.credentials.registerContainer**
+
+First, the registerContainer function allows you to register your extensions implementation of ExtensionCredentialsContainer interface.
+
+Request:
+
+```js
+const credentialsContainer = {
+    create(options) {
+        console.log("Demo Extension: credentials.create");
+        return Promise.resolve(null);
+    },
+    get(options) {
+        console.log("Demo Extension: credentials.get");
+        return Promise.resolve(null);
+    },
+    store(credential) {
+        console.log("Demo Extension: credentials.store");
+        return Promise.resolve();
+    }
+}
+
+browser.credentials.registerContainer(credentialsContainer);
+```
+
+**browser.credentials.isActiveContainer**
+
+This retrieves a boolean indicating if this extension is the active credential manager.
+
+```js
+browser.credentials.isActiveContainer();
```

Comment:
Does this function even make sense to include? Only really makes sense if we allow only one active credential manager.